integration-with-profiles tests failed in CI with "Failed to execute kubectl auth"

integration-with-profiles tests [failed in CI](https://github.com/canonical/training-operator/actions/runs/6159071489/job/16713154061#step:4:120) with the following error
```
FAILED tests/integration/test_charm_with_profile.py::test_authorization_for_creating_resources[examples/tfjob.yaml] - AssertionError: Failed to execute kubectl auth (1): no
```

The error seems intermittent since rerunning the CI fixed it. I created this issue in order to document that we have came across this in the case we stumble upon it again.

### Reproduce
Not sure how to reproduce since the error seems intermittent. 

### Logs
Nothing looks off in the CI logs. These are the last workload logs
```
2023-09-12T12:10:41.199Z [container-agent] 2023-09-12 12:10:41 DEBUG juju.machinelock machinelock.go:202 created rotating log file "/var/log/machine-lock.log" with max size 10 MB and max backups 5
2023-09-12T12:10:41.200Z [container-agent] 2023-09-12 12:10:41 DEBUG juju.machinelock machinelock.go:186 machine lock released for training-operator/0 uniter (run start hook)
2023-09-12T12:10:41.200Z [container-agent] 2023-09-12 12:10:41 DEBUG juju.worker.uniter.operation executor.go:121 lock released for training-operator/0
2023-09-12T12:10:41.200Z [container-agent] 2023-09-12 12:10:41 DEBUG juju.worker.uniter resolver.go:188 no operations in progress; waiting for changes
2023-09-12T12:10:41.200Z [container-agent] 2023-09-12 12:10:41 DEBUG juju.worker.uniter agent.go:20 [AGENT-STATUS] idle:
```
and these the last operator logs
```
2023-09-12T12:10:14.116Z [training-operator] 2023-09-12T12:10:14Z	INFO	Starting workers	{"controller": "mpijob-controller", "worker count": 1}
2023-09-12T12:10:14.920Z [pebble] GET /v1/changes/1/wait?timeout=4.000s 1.002027076s 200
2023-09-12T12:10:27.782Z [pebble] GET /v1/plan?format=yaml 3.65407ms 200
2023-09-12T12:10:39.591Z [pebble] GET /v1/plan?format=yaml 133.702µs 200
```

--- a/tests/integration/test_charm_with_profile.py
+++ b/tests/integration/test_charm_with_profile.py
@@ -2,6 +2,7 @@
 # See LICENSE file for licensing details.
 import glob
 import logging
+import time
 from pathlib import Path
 
 import lightkube
@@ -9,8 +10,17 @@ import pytest
 import yaml
 from lightkube import codecs
 from lightkube.generic_resource import create_global_resource
+from lightkube.resources.core_v1 import ServiceAccount
+from lightkube.resources.rbac_authorization_v1 import RoleBinding
 from pytest_operator.plugin import OpsTest
-from tenacity import RetryError, Retrying, stop_after_attempt, stop_after_delay, wait_exponential
+from tenacity import (
+    RetryError,
+    Retrying,
+    before_sleep_log,
+    stop_after_attempt,
+    stop_after_delay,
+    wait_exponential,
+)
 
 basedir = Path("./").absolute()
 PROFILE_NAMESPACE = "profile-example"
@@ -60,6 +70,7 @@ async def test_authorization_for_creating_resources(
     # Set up for creating an object of kind *Job
     job_yaml = yaml.safe_load(Path(example).read_text())
     training_job = job_yaml["kind"]
+    log.info(f"Checking `kubectl can-i create` for {training_job}")
     _, stdout, _ = await ops_test.run(
         "kubectl",
         "auth",
@@ -112,18 +123,35 @@ def apply_profile(lightkube_client):
 
     # Apply Profile first
     apply_manifests(lightkube_client, PROFILE_FILE_PATH)
+    log.info(f"Profile from {PROFILE_FILE_PATH} applied.")
 
-    # Allow time for the Profile to be created
+    # Ensure the Profile is fully initialized
     try:
         for attempt in Retrying(
             stop=(stop_after_attempt(10) | stop_after_delay(30)),
             wait=wait_exponential(multiplier=1, min=5, max=10),
             reraise=True,
+            before_sleep=before_sleep_log(log, logging.INFO),
         ):
             with attempt:
-                lightkube_client.get(profile_resource, name=PROFILE_NAME)
+                # Look for the Profile and some of the objects expected in an instantiated
+                # Profile (to avoid https://github.com/canonical/training-operator/issues/136)
+                for resource, name, namespace in (
+                    (profile_resource, PROFILE_NAME, None),
+                    (ServiceAccount, "default-editor", PROFILE_NAME),
+                    (RoleBinding, "default-editor", PROFILE_NAME),
+                ):
+                    log.info(f"Looking for {resource} of name {name}")
+                    lightkube_client.get(resource, name=name, namespace=namespace)
+                    log.info(f"Found {resource} of name {name}")
     except RetryError:
-        log.info(f"Profile {PROFILE_NAME} not found.")
+        log.info(f"Profile {PROFILE_NAME} not found or found to be incomplete.")
+
+    # Wait a few seconds more just in case the profile-controller is still creating the Profile
+    # to avoid https://github.com/canonical/training-operator/issues/136
+    sleeptime = 5
+    log.info(f"Waiting {sleeptime}s to ensure Profile is fully initialized")
+    time.sleep(sleeptime)
 
     yield
 

